Source/Provers/SMTLib/SMTLibProcess.cs

```diff
@@ -113,85 +128,60 @@ public override void Send(string cmd)
       toProver.WriteLine(cmd);
     }
 
-    internal Inspector Inspector
-    {
-      get { return inspector; }
-    }
+    internal Inspector Inspector => inspector;
 
-    public override SExpr GetProverResponse()
+    public override async Task<SExpr> GetProverResponse()
     {
-      toProver.Flush();
+      await toProver.FlushAsync();
 
-      while (true)
-      {
-        var exprs = ParseSExprs(true).ToArray();
-        Contract.Assert(exprs.Length <= 1);
-        if (exprs.Length == 0)
-        {
+      while (true) {
+        var exprs = await ParseSExprs(true).ToListAsync();
+        Contract.Assert(exprs.Count <= 1);
+        if (exprs.Count == 0) {
           return null;
         }
 
         var resp = exprs[0];
-        if (resp.Name == "error")
-        {
-          if (resp.Arguments.Length == 1 && resp.Arguments[0].IsId)
-          {
-            if (resp.Arguments[0].Name.Contains("max. resource limit exceeded"))
-            {
+        if (resp.Name == "error") {
+          if (resp.Arguments.Length == 1 && resp.Arguments[0].IsId) {
+            if (resp.Arguments[0].Name.Contains("max. resource limit exceeded")) {
               return resp;
-            }
-            else {
+            } else {
               HandleError(resp.Arguments[0].Name);
               return null;
             }
-          }
-          else {
+          } else {
             HandleError(resp.ToString());
             return null;
           }
-        }
-        else if (resp.Name == "progress")
-        {
-          if (inspector != null)
-          {
+        } else if (resp.Name == "progress") {
+          if (inspector != null) {
             var sb = new StringBuilder();
-            foreach (var a in resp.Arguments)
-            {
-              if (a.Name == "labels")
-              {
+            foreach (var a in resp.Arguments) {
+              if (a.Name == "labels") {
                 sb.Append("STATS LABELS");
-                foreach (var x in a.Arguments)
-                {
+                foreach (var x in a.Arguments) {
                   sb.Append(" ").Append(x.Name);
                 }
-              }
-              else if (a.Name.StartsWith(":"))
-              {
+              } else if (a.Name.StartsWith(":")) {
                 sb.Append("STATS NAMED_VALUES ").Append(a.Name);
-                foreach (var x in a.Arguments)
-                {
+                foreach (var x in a.Arguments) {
                   sb.Append(" ").Append(x.Name);
                 }
-              }
-              else
-              {
+              } else {
```

Comment:
I would avoid such refactorings in general, since they pollute the Git history.  Incidentally, this whole method  would probably look a lot better with pattern matching.

Reply:
Agreed, I'm not sure how this happened. The IDE reformatted this method and I accidentally committed it.